### PR TITLE
trim non-letter from beginning of statement in preview

### PR DESCRIPTION
**Pull request by acharis**
this allows us to analyze:
(select id from foo) union all (select id from bar) limit 10;

Signed-off-by: Alex Charis <acharis@hubspot.com>

### CODE REVIEW

File: go/vt/sqlparser/analyzer.go [hunk 1]
```diff
@@ -60,6 +60,8 @@ func Preview(sql string) int {
 		firstWord = trimmed[:end]
 	}
 
+	firstWord = strings.TrimFunc(firstWord, func(r rune) bool { return !unicode.IsLetter(r) })
+
```
Comment (sougou) [inline go/vt/sqlparser/analyzer.go]: I think this function makes the previous `IndexFunc` call unnecessary. Let's remove it.

Comment (acharis) [reply go/vt/sqlparser/analyzer.go]: done

**Comment by sougou** [comment]
Ok. That was bad advice. All tests are broken now. I think the more correct thing to do would have been to move your TrimFunc before the IndexFunc call. Also, I think it should be TrimLeftFunc.

**Comment by acharis** [comment]
apparently i didn't sign-off the revert, so i just did a force push (after rebase-squash)

**Comment by acharis** [comment]
tests are still failing...want to come up with something else?

**Pull request merged**